### PR TITLE
stderr from Stage#run is captured

### PR DESCRIPTION
**Pull request by akiellor**
fix of #33

### CODE REVIEW

**Comment by quad** [comment]
Why the collapse into `sh -c`? It was implemented the other way as a hedge against string quoting or spacing bugs.

(Not that I'm implying any were introduced. I'm just not certain! Should probably add some tests around that.)

File: lib/bt/stage.rb [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+      @opts = {:out => $stdout}.merge(opts)
+      if opts[:silent]
+        @opts[:out] = IO.pipe[1]
+      end
+    end
+
+    def execute
+      result = ''
+      exitstatus = nil
+      Open3.popen2e('sh -') do |input, output, wait_thread|
```
Comment (quad) [inline lib/bt/stage.rb]: Does simplifying this back down to a `popen` pass the tests

File: lib/bt/stage.rb [hunk 2]
```diff
@@ -1,5 +1,35 @@
 module BT
   require 'bt/yaml'
+  require 'open3'
+
+  class Command < Struct.new :command
+    def initialize command, opts = {}
+      super(command)
+      @opts = {:out => $stdout}.merge(opts)
```
Comment (quad) [inline lib/bt/stage.rb]: Where are options being used?

Comment (akiellor) [inline lib/bt/stage.rb]: Simplified this down to a simple flag. The silent flag is used for testing to stop output to stdout.

File: lib/bt/stage.rb (hunk 1 above)
Comment (akiellor) [inline lib/bt/stage.rb]: Switch back to the popen usage...tests still pass.

**Comment by quad** [comment]
This is good shit.

**Pull request merged**